internal/publish/publish_test.go

```diff
@@ -198,54 +195,21 @@ func (s *PublishSuite) TestPublishWithClientRedeployErrors() {
 	target := deployment.New()
 	target.ID = "myContentID"
 
+	// Fail the preflight existing content checks. Further testing of these will
+	// occur in the connect_client package
+
```

Comment:
The thought with these changes is that the real testing has moved into the `client` implementation (`client_connect`). What remains here is enough testing to know if the capabilities check fails with an error, then this method `publishWithClient` will fail with an error.